internal/smartlink/smartlink.go

```diff
@@ -55,6 +56,15 @@ func Link(src, dest string) error {
 			return errs.Wrap(err, "could not read directory %s", src)
 		}
 		for _, entry := range entries {
+			circular, err := isCircularLink(filepath.Join(src, entry.Name()))
+			if err != nil {
+				return errs.Wrap(err, "could not check for circular link")
+			}
+			if circular {
+				multilog.Error("Build contains a circular symlink: %s -> %s", filepath.Join(src, entry.Name()), dest)
+				continue
```

Comment:
I think we should error out on this, unless BE has a solution? Continuing would just give you a corrupted runtime.

Reply:
BE considers this an error case and has patched it, but advised that it could happen again so we should guard against it.

I'm not sure how this comment works with your next one. We should error our if we encounter a circular link but also support it?